services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/resource_tracker_pricing_plans.py

```diff
@@ -0,0 +1,90 @@
+from typing import Annotated
+
+from fastapi import Depends, Query
+from models_library.api_schemas_webserver.resource_usage import (
+    PricingDetailMinimalGet,
+    PricingPlanGet,
+)
+from models_library.products import ProductName
+from models_library.services import ServiceKey, ServiceVersion
+
+from ..api.dependencies import get_repository
+from ..modules.db.repositories.resource_tracker import ResourceTrackerRepository
+
+
+async def list_pricing_plans(
+    product_name: ProductName,
+    resource_tracker_repo: Annotated[
+        ResourceTrackerRepository, Depends(get_repository(ResourceTrackerRepository))
+    ],
+    service_key: ServiceKey = Query(None),
+    service_version: ServiceVersion = Query(None),
+) -> list[PricingPlanGet]:
+    list_of_pricing_plan_get = []
+    if service_key is None and service_version is None:
```

Comment:
I would split this if else into two separate private functions to make it more redable